server/src/test/java/org/elasticsearch/common/util/MapsTests.java

```diff
@@ -123,6 +127,51 @@ public void testDeepEquals() {
         assertFalse(Maps.deepEquals(map, mapModified));
     }
 
+    public void testCollectToUnmodifiableSortedMap() {
+        SortedMap<String, String> canadianProvinces = Stream.of(new Tuple<>("ON", "Ontario"),
+                new Tuple<>("QC", "Quebec"),
+                new Tuple<>("NS", "Nova Scotia"),
+                new Tuple<>("NB", "New Brunswick"),
+                new Tuple<>("MB", "Manitoba"))
+            .collect(Maps.toUnmodifiableSortedMap(Tuple::v1, Tuple::v2));
+
+        assertThat(canadianProvinces, equalTo(new TreeMap<>(Maps.ofEntries(
+            Stream.of(new AbstractMap.SimpleEntry<>("ON", "Ontario"),
+                    new AbstractMap.SimpleEntry<>("QC", "Quebec"),
+                    new AbstractMap.SimpleEntry<>("NS", "Nova Scotia"),
+                    new AbstractMap.SimpleEntry<>("NB", "New Brunswick"),
+                    new AbstractMap.SimpleEntry<>("MB", "Manitoba"))
+                .collect(Collectors.toList())
+        ))));
+        expectThrows(UnsupportedOperationException.class, () -> canadianProvinces.put("BC", "British Columbia"));
+    }
+
+    public void testCollectRandomListToUnmodifiableSortedMap() {
+        List<Tuple<String, String>> tuples = randomList(1, 100,
```

Comment:
I would normally guard against duplicates here to be sure the test never fails unwarranted (though unlikely with 10 char random keys). So perhaps collect these in a set, convert to list and do `Collections.shuffle(randomOrderQuantities, random())` on the list afterwards.